Balancer-Bots/Stable-Pool-Depegging/src/utils.ts

```diff
@@ -13,29 +15,109 @@ export interface NetworkData {
 export const toBn = (ethersBn: ethers.BigNumber) => new BigNumber(ethersBn.toString());
 
 export type GetAmpUpdateStartedLog = (
-  blockNumber: number
+  txEvent: TransactionEvent
 ) => Promise<(ethers.utils.LogDescription & { emitter: string })[]>;
 
-export function provideGetAmpUpdateStartedLogs(
-  networkManager: NetworkManager<NetworkData>,
-  provider: ethers.providers.Provider
-): GetAmpUpdateStartedLog {
-  const stablePoolIface = new ethers.utils.Interface([AMP_UPDATE_STARTED_ABI]);
-  const topics = [stablePoolIface.getEventTopic("AmpUpdateStarted")];
-
-  return async (blockNumber: number) => {
-    const encodedLogs = await provider.getLogs({
-      topics,
-      fromBlock: blockNumber,
-      toBlock: blockNumber,
-    });
+export function provideGetAmpUpdateStartedLogs(networkManager: NetworkManager<NetworkData>): GetAmpUpdateStartedLog {
+  return async (txEvent: TransactionEvent) => {
+    const encodedLogs = txEvent.filterLog([AMP_UPDATE_STARTED_ABI], networkManager.get("stablePoolAddresses"));
 
     const stablePoolAddresses = networkManager.get("stablePoolAddresses");
 
     return encodedLogs
       .filter((log) => stablePoolAddresses.some((address) => address.toLowerCase() === log.address.toLowerCase()))
```

Comment:
Not 100% sure on this, but I think now that you're passing the addresses to the `.filterLog`, this filtering is not needed

Reply:
that's right. good catch 👍